### PR TITLE
chore/ci: use cargo lipo to generate universal iOS fat libs

### PR DESCRIPTION
**Pull request by ravinderjangra**

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -216,7 +175,7 @@ jobs:
   deploy:
     name: Deploy
     runs-on: ubuntu-latest
-    needs: [build, build-android, build-ios, build-ios-universal]
+    needs: [build, build-android, build-ios, build-ios]
```
Comment (m-cat) [inline .github/workflows/master.yml]: Looks like we have `build-ios` twice here

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -120,79 +127,31 @@ jobs:
...
-
-      # Build dev libraries.
-      - uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: >
-            --release --manifest-path=safe_authenticator/Cargo.toml
-            --features=mock-network --target=${{ matrix.target }}
+      
+      # Install cargo-lipe for universal iOS libs.
```
Comment (m-cat) [inline .github/workflows/master.yml]:
```suggestion
      # Install cargo-lipo for universal iOS libs.
```

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -108,75 +115,31 @@ jobs:
         with:
           path: target
           key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
-
-      # Build prod libraries.
+      
+      # Install cargo-lipe for universal iOS libs.
```
Comment (m-cat) [inline .github/workflows/pr.yml]:
```suggestion
      # Install cargo-lipo for universal iOS libs.
```

File: .github/workflows/master.yml (hunk 1 above)
Comment (ravinderjangra) [reply .github/workflows/master.yml]: will update this

File: .github/workflows/master.yml (hunk 2 above)
Comment (ravinderjangra) [reply .github/workflows/master.yml]: will update this

**Pull request merged**